met-api/src/met_api/models/engagement.py

```diff
@@ -141,6 +141,19 @@ def update_engagement(cls, engagement: EngagementSchema) -> DefaultMethodResult:
         db.session.commit()
         return DefaultMethodResult(True, 'Engagement Updated', engagement_id)
 
+    @classmethod
+    def edit_engagement(cls, engagement_data: dict) -> DefaultMethodResult:
```

Comment:
Just a quick review..
Let model return the model .We are trying to move away from DefaultMethodResult..We will remove it eventually..
If not record: can return a None
If its an error , it shud throw an error

Reply:
Got it, will do